sources/Inputs/include/NetworkManager.h

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright (c) 2020, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+
+/**
+ * @file  NetworkManager.h
+ *
+ * @brief Network manager header file
+ *
+ */
+
+#pragma once
+
+#include "Node.h"
+
+#include <DYNDataInterface.h>
+#include <map>
+
+namespace dfl {
+namespace inputs {
+
+/**
+ * @brief Network manager, handling the network input file
+ *
+ * Relies on DYNAWO data interface
+ */
+class NetworkManager {
+ public:
+  /**
+  * @brief Constructor
+  *
+  * @param filepath network file path
+  */
+  explicit NetworkManager(const std::string& filepath);
+
+ private:
+  /**
+   * @brief Build node tree from data interface
+   */
+  void buildTree();
+
+ private:
+  boost::shared_ptr<DYN::DataInterface> interface_;        ///< data interface
+  std::map<Node::NodeId, boost::shared_ptr<Node>> nodes_;  ///< nodes representing the node tree
```

Comment:
Do we really need this to be sorted? Why not using unordered_map?

Reply:
insert is in O(log(N)) while for unordered_map it is O(N). For now, we access only for connection at the beginning but we go through the tree by walking through each node. For me, insert was the relevant operation, not the access

Reply:
average insert for unordered_map is O(1), so if the order of the node is not relevant when visiting the tree unordered_map is definitively the best option

Reply:
or maybe even a vector if research is not required

Reply:
No, we need to perform connections between nodes, it will take too long with a vector

Reply:
OK for unordered_map